spec/models/manageiq/providers/azure/cloud_manager/provision/cloning_spec.rb

```diff
@@ -113,4 +121,34 @@
       expect(provision.create_nic(false)).to eql(@nic_object.id)
     end
   end
+
+  context "start_clone_task" do
+    before do
+      @resource_group = FactoryBot.create(:azure_resource_group)
+      @dest_name = "test"
+      @clone_options = {clone_options: {name: "test-name", location: "germanycentralwest"}}
+
+      allow(Azure::Armrest::Network::IpAddressService).to receive(:new).and_return(ip_address_service)
+      allow(Azure::Armrest::Network::NetworkInterfaceService).to receive(:new).and_return(nic_service)
+      allow(Azure::Armrest::Configuration).to receive(:new).and_return(configuration)
+      allow(Azure::Armrest::VirtualMachineService).to receive(:new).and_return(vms)
+
+      allow(provision).to receive(:resource_group).and_return(@resource_group)
+      allow(provision).to receive(:dest_name).and_return(@dest_name)
+      allow(provision).to receive(:phase_context).and_return(@clone_options)
+      allow(vms).to receive(:create).and_raise(Azure::Armrest::BadRequestException.new('errors', 'details', 'info'))
+
+      allow(ip_address_service).to receive(:get).and_return(public_ip)
+      allow(nic_service).to receive(:get).and_return(network_interface)
+      
+      allow(nic_service).to receive(:delete)
+      allow(ip_address_service).to receive(:delete)
+    end
+  
+    it 'deletes the public IP and network interface when a BadRequestException is raised' do
+      provision.start_clone_task
+      expect(ip_address_service).to have_received(:delete)
+      expect(nic_service).to have_received(:delete)      
+    end
```

Comment:
@liu-samuel can you add a second `it` that checks the phase_context is set properly after BadRequestException is raised ?